tests/unit/invert/test_fft.cxx

```diff
@@ -0,0 +1,149 @@
+#include "gtest/gtest.h"
+
+#include "bout/array.hxx"
+#include "bout/constants.hxx"
+#include "dcomplex.hxx"
+#include "fft.hxx"
+#include "test_extras.hxx"
+
+#include <algorithm>
+#include <iostream>
+#include <numeric>
+
+TEST(FFTTest, rfft) {
+
+  // Make sure fft functions are quiet by setting fft_measure to false
+  bout::fft::fft_init(false);
+
+  constexpr int size{8};
+
+  // Calculate sin(x) + cos(2x) on [0, 2pi]
+  Array<BoutReal> indices(size);
+  std::iota(indices.begin(), indices.end(), 0);
+
+  Array<BoutReal> input(size);
+  std::transform(indices.begin(), indices.end(), input.begin(),
+                 [](BoutReal i) -> BoutReal {
+                   return std::sin(i * TWOPI / size) + std::cos(2. * i * TWOPI / size);
+                 });
+
+  Array<dcomplex> output(size);
```

Comment:
Shouldn't we expect a smaller output than the input (e.g. size/2)? It might be good to not overallocate as that way we can catch attempts to go beyond our expected bounds better.